### ISSUE TITLE
Enhance Metrics to filter based on verbs and URI(s)

### ISSUE THREAD

**Issue by sumantfordev** [Kong/kong-plugin-prometheus#103]
Enhance Metrics to filter based on verbs and URI(s)
Currently the http codes counter metrics, bandwidth and latency metrics are generated per service, and route.
Can this be enhanced to generate metrics per http method( POST/GET etc) and per URI. Like if the route has uri like /users/[*] , then it will create metrics report for all api(s) /users/all or /users/<some user id>
Expectation:
kong_http_status{code="200",service="api-1",route="api-users", verb="GET" Uri="/users?emailid=someemail@domain.com"} 16

**Comment by sumantfordev** [comment]
By unbounded labels you mean about the URI part right, Verbs are still a close set and should not affect the performance right?

**Comment by fffonion** [comment]
I agree with @hbagdi , this use case is more like a logging not metrics. You would probably do that with a ELK stack. Or if you are already familiar with prometheus query syntax, try promtail + loki from Grafana.

**Comment by hbagdi** [comment]
HTTP verbs/methods are open-ended. There are standard methods but there is nothing in the HTTP spec stopping you to define your own `FOO` method.

As Wangchong pointed out, this increasingly sounds like abusing metrics as a substitute for logs and log analysis tools.
